packages/core/src/common/abstractPureComponent.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as React from "react";
+
+/**
+ * A pure abstract component that Blueprint components can extend
+ * in order to add some common functionality like runtime props validation.
+ */
+export abstract class AbstractPureComponent<P, S> extends React.PureComponent<P, S> {
```

Comment:
there must be a way to reuse the code from `AbstractComponent` instead of duplicating it here. maybe this is the impetus to refactor AC into separate validation and timeout controllers.